site-src/concepts/versioning.md

```diff
@@ -1,116 +1,70 @@
-will exclude these fields from stable CRDs. Experimental fields may be removed
-from the API. Due to the experimental nature of these CRDs, they are not
-recommended for production use.
-
-If experimental fields are removed or renamed, the original field name should be
-removed from the go struct, with a tombstone comment ensuring the field name
-will not be reused. 
-
+### API Versions (v1alpha2, v1beta1)
+Within Gateway API, API versions are primarily used to indicate the stability of
+a resource. For example, if a resource has not yet graduated to beta, it is
+still possible that it could either be removed from the API or changed in
+backwards incompatible ways. For more information on API versions, refer to the
+[full Kubernetes API versioning
+documentation](https://kubernetes.io/docs/reference/using-api/#api-versioning).
+
+### Release Channels (experimental, stable)
+Release channels are used to indicate the stability of fields within a resource.
+For example, even if a resource has already graduated to beta, we need a way to
+test out new fields and features before committing to their long term stability.
+With upstream Kubernetes APIs, this is accomplished with [feature
+gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/).
+Unfortunately, CRDs do not have a similar concept yet.
+
+Instead of trying to recreate feature gates in this project, we've introduced
+release channels. Starting in v0.5.0, all new fields and features will start in
+the experimental set of CRDs. From that point, these may graduate to the stable
+release channel or be dropped from the API entirely.
+
+Due to their unstable nature, experimental CRDs should not be used for
+production workloads.
+
+## Version Indicators
 Each CRD will be published with annotations that indicate their bundle version
 and channel:
 
 ```
 gateway.networking.k8s.io/bundle-version: v0.4.0
 gateway.networking.k8s.io/channel: stable|experimental
 ```
+
+## What can Change
+When using or implementing this API, it is important to understand what can
+change across API versions.
+
+### Patch version (v0.4.0 -> v0.4.1)
+* Clarifications to godocs
+* Updates to CRDs and/or code to fix a bug
+* Fixes to typos
+
+### Minor version (v0.4.0 -> v0.5.0)
+* Everything that is valid in a patch release
+* New experimental API fields or resources
+* Loosened validation
+* Making required fields optional
+* Removal of experimental fields
+* Removal of experimental resources
+* Graduation of fields or resources from experimental to stable track
+* Introduction of a new **API version**, which may also include:
+  * Renamed fields
+  * Anything else that is valid in a new Kubernetes API version
+  * Removal/tombstoning of beta fields
+* Removal of an API resource following [Kubernetes deprecation
+  policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/)
+
+### Major version (v0.x to v1.0)
+* Everything that is valid in a patch release
```

Comment:
Better to say: There are no API compatibility guarantees when the major version changes.

Reply:
Updated, thanks!